libraries/stdlib/genosl/mx_normalmap.osl

```diff
@@ -1,16 +1,21 @@
 void mx_normalmap_vector2(vector value, vector2 normal_scale, vector N, vector T, vector B, output vector result)
 {
-    vector decodedValue;
     if (value == vector(0.0))
     {
-        decodedValue = vector(0.0, 0.0, 1.0);
+        result = N;
     }
     else
     {
-        decodedValue = value * 2.0 - 1.0;
+        // The OSL backend uses dPdu and dPdv for tangents and bitangents, but these vectors are not
+        // guaranteed to be orthonormal.
+        //
+        // Orthogonalize the tangent frame using Gram-Schmidt, unlike in the other backends.
+        //
+        vector v = value * 2.0 - 1.0;
+        vector Tn = normalize(T - dot(T, N) * N);
+        vector Bn = normalize(cross(N, Tn));
```

Comment:
Just realized that this line needs to be adjusted to accommodate the explicit bitangent..

Reply:
Ah, that makes sense.  Did you want to make that additional modification, since you have the best visibility on the explicit bitangent use case?

Reply:
(Would appreciate if you could commit/test this, since building OSL is a nightmare on macOS/Windows..)

Reply:
No problem, @pablode, and I'll propose that change from here.  Once we have a proposal that looks correct to everyone, I can run the full suite of GLSL/OSL comparisons for validation.

Reply:
@pablode I've heard from @klucknav at Pixar that this latest change breaks one of the examples in their test suite, though I'm not fully looped in on the details.

Reply:
Scratch that last comment, and I've heard from @klucknav that this latest version works in Pixar tests.  Let's go ahead and merge this for the release candidate tomorrow.